homework/index.js

```diff
@@ -30,17 +15,123 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  // a function to create td without link
+  function createTd(textKey, textContent, tbody) {
+    const parent = createAndAppend('tr', tbody);
+    createAndAppend('td', parent, {
+      text: `${textKey}: `,
+      class: 'label',
+    });
+    createAndAppend('td', parent, {
+      text: textContent,
+    });
+  }
+  // table of repository's basic information
+  function createDetailsTable(repository, container) {
+    const repositoryData = createAndAppend('div', container, { class: 'left-div white-frame' });
+    const table = createAndAppend('table', repositoryData);
+    const tbody = createAndAppend('tbody', table);
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, {
+      text: 'Repository :',
+      class: 'label',
+    });
+    // manually create td with a link
+    const td = createAndAppend('td', tr);
+    createAndAppend('a', td, {
+      href: repository.html_url,
+      text: repository.name,
+      target: '_blank',
+    });
+    createTd('Description', repository.description, tbody);
+    createTd('Forks', repository.forks, tbody);
+    createTd('Updated', repository.updated_at, tbody);
+  }
+
+  // contributors list
+  async function listContributors(repository, container) {
+    const contributorUrl = repository.contributors_url;
+    const response = await fetch(contributorUrl);
```

Comment:
1. All `await` expressions should be inside the `try` block.
2. You need to check for HTTP errors. See my message in the #class21 slack channel.